Make `Checksum` task compatible with Gradle 5.0

bamboo opened the pull request:
By ignoring directories found in `IncrementalTaskInputs`.

This is necessary because Gradle 5.0 includes directories in `IncrementalTaskInputs` when the input is represented as a file tree.

bamboo committed:
Upgrade build to Gradle 5.0 and `plugin-publish` 0.10.0

bamboo committed:
Make `Checksum` compatible with Gradle 5.0

By skipping directories included in the incremental inputs.

wolfs commented on src/main/java/org/gradle/crypto/checksum/Checksum.java:
```diff
@@ -116,7 +119,11 @@ public void execute(InputFileDetails inputFileDetails) {
             public void execute(InputFileDetails inputFileDetails) {
-                getProject().delete(outputFileFor(inputFileDetails.getFile()));
+                File input = inputFileDetails.getFile();
+                if (input.isDirectory()) {
```
We could delete removed directories, right? So we don't have empty directories lying around in the output directory.

I am not feeling strongly about this and I guess removing empty directories would also require extra coverage.

bamboo replied:
The plot thickens. The plugin doesn't actually replicate the input directory tree in the output directory so there are no directories to remove.

wolfs commented on src/functionalTest/groovy/org/gradle/crypto/checksum/IncrementalChecksumSpec.groovy:
```diff
@@ -0,0 +1,73 @@
+            }
+        """
+        newFolder('input')
+        newFile('input/foo.txt') << 'foo'
```
Should we add some files deeper in the directory structure? Note that even with pre-5.0 Gradle versions, the build probably would have failed, given that non-root directories have been reported as changes before as well.

bamboo replied:
Makes sense. Thanks.

wolfs reviewed (approved):
Looks good. I added two comments for further fixes/features. Feel free to address them or open issues for them.

bamboo committed:
Improve test coverage

By adding a subdirectory before the 2nd run which causes the plugin to
fail with previous Gradle versions as well.
